core/workspace_svg.js

```diff
@@ -944,6 +944,7 @@ Blockly.WorkspaceSvg.prototype.loadAudio_ = function(filenames, name) {
     if (ext && audioTest.canPlayType('audio/' + ext[1])) {
       // Found an audio format we can play.
       sound = new window['Audio'](filename);
+      sound.setAttribute('preload', 'auto');
```

Comment:
The preload attribute is an official <audio> attribute, so it will not break anything.
See https://www.w3.org/TR/html5/embedded-content-0.html#attr-media-preload

The former solution has not really worked on iOS:
```
// iOS can only process one sound at a time.  Trying to load more than one		
// corrupts the earlier ones.  Just load one and leave the others uncached.		
...
```

According MDN in most cases the mobile browsers will load the Metadata but not buffer the audio to 100%, which is by design and expected.
That's also the reason why the former solution not really worked on iOS.